MapboxDirections/MBAnnotation.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  MBAnnotationType.swift
+//  MapboxDirections
+//
+//  Created by Bobby Sudekum on 3/20/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+import Foundation
+
+@objc(AnnotationType)
+public enum AnnotationType: Int, CustomStringConvertible {
+
+    case congestion
+    
+    case distance
+    
+    case duration
+    
+    case nodes
```

Comment:
(I'm going to call annotation types "attributes" for the rest of this review; see below.)

Let's call this `openStreetMapNodeIdentifier`, similar to how MapboxGeocoder.swift has a `wikidataItemIdentifier`. If on the other hand we expect the same underlying string to be used for non-OSM data sources, `nodeIdentifier` would be better. (I'd like to avoid using "node" by itself, because perhaps we may associate more attributes with OSM nodes in the future.)